include/cling/Interpreter/Interpreter.h

```diff
@@ -540,6 +540,25 @@ namespace cling {
     ///
     std::string lookupFileOrLibrary(llvm::StringRef file);
 
+    ///\brief Loads a shared library.
+    ///
+    ///\param [in] filename - The file to loaded.
+    ///\param [in] lookup - Whether to try to resolve the filepath
+    ///
+    ///\returns kMoreInputExpected is returned when file could not be found
+    /// otherwise kSuccess or kFailure
+    CompilationResult loadLibrary(const std::string& filename,
+                                  bool lookup = true);
+
+    ///\brief Loads header file
+    ///
+    ///\param [in] filename - The file to loaded.
+    ///\param [out] T -  Transaction containing the loaded file.
+    ///\returns result of the compilation.
+    ///
+    CompilationResult loadHeader(const std::string& filename,
+                               Transaction** T = 0);
```

Comment:
Could you add some tests. This part is going to be heavily used because it is a very visible interface.